oci/layer/tar_extract.go

```diff
@@ -362,6 +362,21 @@ func (te *TarExtractor) ociWhiteout(root string, dir string, file string) error
 	return errors.Wrap(err, "whiteout remove")
 }
 
+func isEmptyDir(name string) (bool, error) {
+	f, err := os.Open(name)
```

Comment:
If you mount ;lowerdir=l1:l2,lowerdir=l3 dest' where l2 has a1/file, and you simply rm -rf dest/a1, then l3 will end up with
```
c--------- 2 root root 0, 0 Jun 13 16:13 a1
```
If instead you 
```
rm -rf dest/a1; mkdir dest/a1; touch dest/otherfile
```
then you'll end up with 
```
drwxrwxr-x 2 root root 4096 Jun 13 16:14 a1
root@serge-l-PF3DENS3 ~/delme/overlay$ ~/test/llistxattrfull ll3/a1
ret 20 on ll3/a1
nret 20 list user.overlay.opaque
```

That's the overlay behavior.  Likewise, OCI should use a .wh_a1 file if deleting a1, and use the .opaque if re-creating a1.  That is how you should tell the difference.

The problem with this isEmptyDir is that my layer may, in fact, rm -rf a1; mkdir a1; and expect a meaningful, empty directory to be there.

Reply:
> mount ;lowerdir=l1:l2,lowerdir=l3 dest
you mean `lowerdir=l1:l2,upperdir=l3 dest` here right?

your third code block is a little munged too, or maybe I am missing something.

SO IIUC, you are saying that we may now be handling these cases correctly:
```
# tar file has:
.wh.mydir
mydir/myfile

# or tar file has
mydir/myfile
.wh.mydir
```
but we will still get this wrong:
```
mydir
.wh.mydir
```
is that right?

I think I agree - we will create the dir, but it will be an empty dir and the isEmptyDir check will return true, which will have us RemoveAll the dir, and it won't exist anymore...

Reply:
Right.  And someone should verify that we are creating the correct OCI whiteout in all cases:

1. If file is being replaced with a new file, no whiteout
2. If file is being deleted, then .wh_
3. If directory is being replaced with new one, than .wh.*opaque
4. If directory is being deleted, then .wh_

And then our conversion *back* to overlay should handle those correctly.

Reply:
> you mean `lowerdir=l1:l2,upperdir=l3 dest` here right?

yeah :)